twitterMini/Services/TwitterDataManager.swift

```diff
@@ -46,26 +63,43 @@ class TwitterDataManager {
             if let error = result.error {
                 completion(.Failure(error))
             } else {
-                // FIXME: use genome
-                if let rawStatuses = result.value {
-                    for statusDictionary in rawStatuses ?? [] {
-                        if let createdAt = statusDictionary["created_at"] as? String,
-                            let text = statusDictionary["text"] as? String,
-                            let likes = statusDictionary["favorite_count"] as? Int,
-                            let retweets = statusDictionary["retweet_count"] as? Int,
-                            let userDictionary = statusDictionary["user"] as? [String: AnyObject],
-                            let name = userDictionary["name"] as? String,
-                            let screenName = userDictionary["screen_name"] as? String,
-                            let profileImageURL = userDictionary["profile_image_url_https"] as? String {
-                            
-                            let user = TwitterUser(screenName: screenName, name: name, profileImageURL: profileImageURL)
-                            let status = TwitterStatus(createdAt: createdAt, text: text, user: user, likes: "\(likes)", retweets: "\(retweets)")
-                            self.statuses.append(status)
-                        }
-                    }
+                if let value = result.value,
+                    let rawStatuses = value {
+                    self.statuses = self.createStatuses(fromDictionaries: rawStatuses)
                 }
                 completion(.Success())
             }
         }
     }
+    
+    func createStatuses(fromDictionaries rawStatuses: [[String:AnyObject]]) -> [TwitterStatus] {
+        // FIXME: use genome
+        var newStatuses = [TwitterStatus]()
+        for statusDictionary in rawStatuses {
+            if let createdAt = statusDictionary["created_at"] as? String,
+                let text = statusDictionary["text"] as? String,
+                let likes = statusDictionary["favorite_count"] as? Int,
+                let retweets = statusDictionary["retweet_count"] as? Int,
+                let userDictionary = statusDictionary["user"] as? [String: AnyObject],
+                let name = userDictionary["name"] as? String,
+                let screenName = userDictionary["screen_name"] as? String,
+                let profileImageURL = userDictionary["profile_image_url_https"] as? String {
+                
+                var mediaURL : String? = nil
```

Comment:
var mediaURL: String?   is all you need.  it's set to nil by default when optional.